// sfmc/index.js
"use strict";
var convert         = require('xml-js');
var request         = require('request');

var sfmc = module.exports = function sfmc() {};

sfmc.prototype.getToken = function(authEndpoint,clientId,clientSecret,refreshToken,accessType){
    // Return new promise
    return new Promise(function(resolve, reject) {
            var payload =   {
                "clientId": clientId,
                "clientSecret": clientSecret,
                "refreshToken": refreshToken,
                "accessType": accessType
            };

            var restoptions = {
                "url":        "https://auth.exacttargetapis.com/v1/requestToken",
                "method":     "POST",
                "headers":    {
                    "content-type":"application/json"
                },
                "body": JSON.stringify(payload)
            };

        	// Do async job
            request.post(restoptions, function(err, response, body) {
                if (err) {
                    console.log(JSON.stringify(err));
                    reject(err);
                } else {
                    //console.log("Access Token",body);
                    var resp = JSON.parse(body);
                    if (resp.accessToken){
                        resolve(resp);
                    } else {
                        reject(new Error('Failed to get access token from SFMC'));
                    }
                }
            })
    });
}

sfmc.prototype.retrieve = function(options) {
    var endpoint = options.Endpoint + 'Service.asmx';

    // Get Data Extension Fields
    var soap =     `<Envelope xmlns="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
                     <Header>
                          <fueloauth>${options.Token}</fueloauth>
                       </Header>
                     <Body>
                          <RetrieveRequestMsg xmlns="http://exacttarget.com/wsdl/partnerAPI">
                             <RetrieveRequest>
                                <ObjectType>${options.ObjectType}</ObjectType>`;

    if (options.ObjectType.toLowerCase().indexOf('dataextensionobject') !== -1){
        for (var p = 0; p < options.Properties.length; p++) {
            soap += "<Properties>" + options.Properties[p] + "</Properties>\n";
        }
    } else {
        let schema = getObjectSchema(options.ObjectType);

        for (var p = 0; p < schema.properties.length; p++) {
            soap += "<Properties>" + schema.properties[p].Name + "</Properties>\n";
        }
    }

    if(options.Filter){
        soap += `            <Filter xsi:type="SimpleFilterPart">
                                       <Property>${options.Filter.Property}</Property>
                                       <SimpleOperator>${options.Filter.SimpleOperator}</SimpleOperator>
                                       <Value>${options.Filter.Value}</Value>
                                    </Filter>`;
    }
    soap +=    `         </RetrieveRequest>
                          </RetrieveRequestMsg>
                     </Body>
                    </Envelope>`;

    //console.log(soap);

    var soapoptions = {
        "url":        endpoint,
        "method":     "POST",
        "headers":    {
            "Content-Type":'text/xml',
            "SoapAction":"Retrieve"
        },
        "body": soap
    };

    console.log('Soap',soap);

    // Return new promise
    return new Promise(function(resolve, reject) {
    	// Do async job
        request.post(soapoptions, function(err, response, body) {
            if (err) {
                console.log('RejectingSoap',err);
                reject(err);
            } else {
                try {
                    console.log('SoapResponse',body);
                    var myObject = parseSoap(body);

                    console.log("Response",JSON.stringify(myObject));

                    resolve(myObject);
                } catch (e) {
                    reject(e)
                }
            }
        })
    })
};

sfmc.prototype.upsert = function(options) {

    var endpoint = 'https://webservice.';
    if (options.stack.toLowerCase() !== 's1'){
        endpoint += options.stack + '.';
    }

    endpoint += 'exacttarget.com/Service.asmx';

    // Get Data Extension Fields
    var soap =     `<Envelope xmlns="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
                        <Header>
                            <fueloauth>${options.token}</fueloauth>
                        </Header>
                        <Body>
                            <UpdateRequest xmlns="http://exacttarget.com/wsdl/partnerAPI">
                                <Options>
                                  <SaveOptions>
                                     <SaveOption>
                                        <PropertyName>DataExtensionObject</PropertyName>
                                        <SaveAction>UpdateAdd</SaveAction>
                                     </SaveOption>
                                  </SaveOptions>
                                </Options>
                                <Objects xsi:type="DataExtensionObject">
                                    <CustomerKey>${options.customerkey}</CustomerKey>
                                    <Keys>`;

                                for (var k in options.keys) {
                                    soap += `<Key><Name>${k}</Name><Value>${options.keys[k]}</Value></Key>`;
                                }

                                soap += "</Keys><Properties>";

                                for (var p in options.properties) {
                                    soap += `<Property><Name>${p}</Name><Value>${options.properties[p]}</Value></Property>`;
                                }

                            soap += "</Properties></Objects></UpdateRequest></Body></Envelope>";

    //console.log(soap);

    var soapoptions = {
        "url":        endpoint,
        "method":     "POST",
        "headers":    {
            "Content-Type":'text/xml',
            "SoapAction":"Update"
        },
        "body": soap
    };

    return new Promise(function(resolve, reject) {
        request.post(soapoptions, function(err, response, body) {
            if (err) {
                reject(err);
            } else {
                try {
                    var myObject = parseSoap(body);
                    resolve(myObject);
                } catch (e) {
                    reject(e)
                }
            }
        })
    })
};

sfmc.prototype.createDataExtension = function(options) {
    //console.log(options);

    var endpoint = 'https://webservice.';
    if (options.Stack.toLowerCase() !== 's1'){
        endpoint += options.Stack + '.';
    }

    endpoint += 'exacttarget.com/Service.asmx';

    // Get Data Extension Fields
    var soap =  `<Envelope xmlns="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
                    <Header>
                        <fueloauth>${options.Token}</fueloauth>
                    </Header>
                    <Body>
                        <CreateRequest xmlns="http://exacttarget.com/wsdl/partnerAPI">
                            <Options/>
                            <Objects xsi:type="ns2:DataExtension" xmlns:ns2="http://exacttarget.com/wsdl/partnerAPI">
                                <Name>${options.definition.Name}</Name>
                                <CustomerKey>${options.definition.CustomerKey}</CustomerKey>
                                <IsSendable>${options.definition.IsSendable}</IsSendable>
                                <IsTestable>${options.definition.IsTestable}</IsTestable>
                                <CategoryID>${options.definition.CategoryID}</CategoryID>`;

    if(options.definition.IsSendable == "true"){

        soap += `               <SendableDataExtensionField>
                                    <Name>${options.definition.SendableDataExtensionField.Name}</Name>
                                </SendableDataExtensionField>
                                <SendableSubscriberField>
                                    <Name>Subscriber Key</Name>
                                </SendableSubscriberField>
        `;
    }

    soap += `                   <Fields>`;

    let field;
    let systemProps = ["StorageType","ModifiedDate","CreatedDate","CustomerKey","ObjectID"];
    for (var f = 0; f < options.definition.fields.length; f++) {
        field = options.definition.fields[f];
        soap += '                   <Field>';

        for (var p in field) {
            if (typeof field[p] !== "object" && systemProps.indexOf(p) == -1) {
                soap += '               <' + p + '>' + field[p] + '</' + p + '>';
            }
        }

        soap += '                   </Field>';
    }

    soap +=    `                </Fields>
                            </Objects>
                        </CreateRequest>
                    </Body>
                </Envelope>`;


    //console.log(soap);


    return new Promise(function(resolve, reject) {
    	// Do async job
        var soapoptions = {
            "url":        endpoint,
            "method":     "POST",
            "headers":    {
                "Content-Type":'text/xml',
                "SoapAction":"Create"
            },
            "body": soap
        };
        request(soapoptions, function(err, response, body) {
            if (err) {
                reject(err);
            } else {
                let parsed;
                try {
                    parsed = clean(JSON.parse(convert.xml2json(response.body, {compact: true, spaces: 4})));

                    let newId = parsed["soap:Envelope"]["soap:Body"].CreateResponse.Results.NewObjectID;
                    resolve(newId);
                } catch (e){
                    reject(e);
                }
            }
        })
    })



    return soap;
};

sfmc.prototype.getEventDefinition = function(id,token) {

    var restoptions = {
        "url":        "https://www.exacttargetapis.com/interaction/v1/eventDefinitions/" + id,
        "method":     "GET",
        "headers":    {
            "authorization":'Bearer ' + token,
            "content-type":"application/json"
        }
    };

    // Return new promise
    return new Promise(function(resolve, reject) {
    	// Do async job
        request.get(restoptions, function(err, response, body) {
            if (err) {
                reject(err);
            } else {
                resolve(JSON.parse(body));
            }
        })
    })
};

sfmc.prototype.getUsers = function(token,mid) {
    // todo - currently limited to 50 users need to add paging support

    var restoptions = {
        "url":        "https://www.exacttargetapis.com/platform/v1/accounts/" + mid + "/users",
        "method":     "GET",
        "headers":    {
            "authorization":'Bearer ' + token,
            "content-type":"application/json"
        }
    };

    // Return new promise
    return new Promise(function(resolve, reject) {
    	// Do async job
        request.get(restoptions, function(err, response, body) {
            if (err) {
                reject(err);
            } else {
                resolve(JSON.parse(body));
            }
        })
    })
};

sfmc.prototype.restRequest = function(options) {
    var qs = '';
    if (options.parameters){
        qs = Object.keys(options.parameters).map(key => key + '=' + options.parameters[key]).join('&');
    }

    let url = "https://www.exacttargetapis.com" + options.path;

    if (qs.length > 0){
        url += "?" + qs;
    }

    let restoptions = {
        "url":        url,
        "method":     options.method,
        "headers":    {
            "authorization":'Bearer ' + options.token,
            "content-type":"application/json"
        }
    };

    if (options.body){
        restoptions.body = options.body;
    }

    //console.log("RestOptions",restoptions);

    // Return new promise
    return new Promise(function(resolve, reject) {
    	// Do async job
        request(restoptions, function(err, response, body) {
            if (err) {
                reject(err);
            } else {
                resolve(JSON.parse(body));
            }
        })
    })
};

function getObjectSchema (objectName){
    var schema = soapSchema();
    for (var i = 0; i < schema.objects.length; i++) {
        if (schema.objects[i].objectname == objectName){
            return schema.objects[i];
        }
    }

    return null;
}

function parseSoap(xml){
    try {
        var results;

        xml = xml.replace(/soap\:/g,'');
        xml = xml.replace(/wsa\:/g,'');
        xml = xml.replace(/wsse:\:/g,'');
        xml = xml.replace(/wsu:\:/g,'');
        var o = JSON.parse(convert.xml2json(xml, {compact: true, spaces: 4}));
        o = clean(o,'');

        if (o.Envelope.Body.RetrieveResponseMsg.Results){
            if(o.Envelope.Body.RetrieveResponseMsg.Results.constructor === Array){
                return o.Envelope.Body.RetrieveResponseMsg.Results;
            } else {
                return [o.Envelope.Body.RetrieveResponseMsg.Results];
            }
        } else {
            return null;
        }

    } catch (e){
        return null;
    }
}

function clean(obj,stack) {
    for (var property in obj) {
        if (obj.hasOwnProperty(property)) {
            if (typeof obj[property] == "object") {
                if (obj[property]["_text"]){
                    obj[property] = obj[property]["_text"];
                    delete obj[property]["_text"];
                }

                if (obj[property]["_attributes"]){
                    delete obj[property]["_attributes"];
                }

                if (obj[property]["_declaration"]){
                    delete obj[property]["_declaration"];
                }
                obj[property] = clean(obj[property], stack + '.' + property);
            }
        }
    }

    return obj;
}

function soapSchema(){
    return {
    	"objects": [{
    		"objectname": "Account",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "AccountType",
    			"FieldType": "Text",
    			"SoapType": "AccountTypeEnum",
    			"IsComplexType": true
    		}, {
    			"Name": "ParentID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "BrandID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "PrivateLabelID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ReportingParentID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Email",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "FromName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "BusinessName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Phone",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Address",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Fax",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "City",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "State",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Zip",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Country",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsActive",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "IsTestAccount",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ClientID1",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "DBID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "DeletedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "EditionID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.PartnerClientKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "InheritAddress",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "UnsubscribeBehavior",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "IsTrialAccount",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.EnterpriseID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentAccount.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentAccount.Name",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentAccount.ParentID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentAccount.CustomerKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentAccount.AccountType",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Locale.LocaleCode",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "TimeZone.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "TimeZone.Name",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Roles",
    			"FieldType": "Text",
    			"SoapType": "Role[]",
    			"IsComplexType": true
    		}, {
    			"Name": "ContextualRoles",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ObjectState",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "LanguageLocale.LocaleCode",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "IndustryCode",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "AccountState",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "AccountUser",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "AccountUserID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "UserID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Email",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "MustChangePassword",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "ActiveFlag",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "ChallengePhrase",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ChallengeAnswer",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsAPIUser",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "NotificationEmailAddress",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.PartnerClientKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Password",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Locale.LocaleCode",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "TimeZone.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "TimeZone.Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "DefaultBusinessUnit",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "LanguageLocale.LocaleCode",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ModifiedBy",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "AccountUserAccount",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "AccountUser.AccountUserID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "AccountUser.Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 128,
    			"IsComplexType": false
    		}, {
    			"Name": "AccountUser.Email",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 254,
    			"IsComplexType": false
    		}, {
    			"Name": "AccountUser.UserID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 254,
    			"IsComplexType": false
    		}, {
    			"Name": "AccountUser.ActiveFlag",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Account.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Account.Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 128,
    			"IsComplexType": false
    		}, {
    			"Name": "Account.IsActive",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Account.CustomerID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "Activity",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Program.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "Task.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "Sequence",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Definition.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Definition.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "PartnerAPIObjectTypeID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Definition",
    			"FieldType": "Text",
    			"SoapType": "APIObject",
    			"IsComplexType": true
    		}, {
    			"Name": "IsActive",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.CreatedBy",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ModifiedBy",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.EnterpriseID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "ActivityInstance",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "ActivityDefinition.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "ProgramInstance.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "TaskInstance.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "SequenceID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Status",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "IsActive",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Definition.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Definition.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "PartnerAPIObjectTypeID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Definition",
    			"FieldType": "Text",
    			"SoapType": "APIObject",
    			"IsComplexType": true
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.CreatedBy",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ModifiedBy",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "StatusLastUpdate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "StatusMessage",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.EnterpriseID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "ProgramID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "BusinessRuleList",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Feature",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Data",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Quality",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsInheritedFromParent",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "IsViewable",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "DisplayName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ProductCode",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "BusinessUnit",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "AccountType",
    			"FieldType": "Text",
    			"SoapType": "AccountTypeEnum",
    			"IsComplexType": true
    		}, {
    			"Name": "ParentID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "BrandID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "PrivateLabelID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ReportingParentID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Email",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "FromName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "BusinessName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Phone",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Address",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Fax",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "City",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "State",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Zip",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Country",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsActive",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "IsTestAccount",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "DBID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "DeletedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "EditionID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "IsTrialAccount",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Locale.LocaleCode",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.EnterpriseID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Subscription.SubscriptionID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Subscription.HasPurchasedEmails",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Subscription.EmailsPurchased",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Subscription.Period",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Subscription.AccountsPurchased",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Subscription.LPAccountsPurchased",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Subscription.DOTOAccountsPurchased",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Subscription.BUAccountsPurchased",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Subscription.AdvAccountsPurchased",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Subscription.BeginDate",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Subscription.EndDate",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Subscription.Notes",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Subscription.ContractNumber",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Subscription.ContractModifier",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.PartnerClientKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentAccount.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentAccount.Name",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentAccount.ParentID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentAccount.CustomerKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentAccount.AccountType",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "DefaultSendClassification.ObjectID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "DefaultHomePage.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "MasterUnsubscribeBehavior",
    			"FieldType": "Text",
    			"SoapType": "UnsubscribeBehaviorEnum",
    			"IsComplexType": true
    		}, {
    			"Name": "InheritAddress",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Roles",
    			"FieldType": "Text",
    			"SoapType": "Role[]",
    			"IsComplexType": true
    		}, {
    			"Name": "SubscriberFilter",
    			"FieldType": "Text",
    			"SoapType": "FilterPart",
    			"IsComplexType": true
    		}, {
    			"Name": "ContextualRoles",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "LanguageLocale.LocaleCode",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "ContentArea",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "RowObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "CategoryID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Layout",
    			"FieldType": "Text",
    			"SoapType": "LayoutType",
    			"IsComplexType": true
    		}, {
    			"Name": "IsDynamicContent",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Content",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsSurvey",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "IsBlank",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Key",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "DataExtension",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsSendable",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "IsTestable",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "SendableDataExtensionField.Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "SendableSubscriberField.Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Template.CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CategoryID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Status",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsPlatformObject",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "DataRetentionPeriodLength",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "DataRetentionPeriodUnitOfMeasure",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "RowBasedRetention",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "ResetRetentionPeriodOnImport",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "DeleteAtEndOfRetentionPeriod",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "RetainUntil",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "DataExtensionField",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "DefaultValue",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "MaxLength",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "IsRequired",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Ordinal",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "IsPrimaryKey",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "FieldType",
    			"FieldType": "Text",
    			"SoapType": "DataExtensionFieldType",
    			"IsComplexType": true
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Scale",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "DataExtension.CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "StorageType",
    			"FieldType": "Text",
    			"SoapType": "DataExtensionFieldStorageType",
    			"IsComplexType": true
    		}]
    	}, {
    		"objectname": "DataExtensionTemplate",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsSendable",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "IsTestable",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "SendableCustomObjectField",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "SendableSubscriberField",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "DataRetentionPeriodLength",
    			"FieldType": "Text",
    			"SoapType": "Int16",
    			"IsComplexType": true
    		}, {
    			"Name": "DataRetentionPeriodUnitOfMeasure",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "RowBasedRetention",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "ResetRetentionPeriodOnImport",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "DeleteAtEndOfRetentionPeriod",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "RetainUntil",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "DataFolder",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentFolder.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentFolder.CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentFolder.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "ParentFolder.Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentFolder.Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentFolder.ContentType",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentFolder.IsActive",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentFolder.IsEditable",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "ParentFolder.AllowChildren",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ContentType",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsActive",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "IsEditable",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "AllowChildren",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ModifiedBy",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.EnterpriseID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.CreatedBy",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "Edition",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsActive",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ProductCode",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsBundle",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "ManualConfig",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "EditionBusinessRule",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "EditionID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "BusinessRuleID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Data",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Quality",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Feature",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsInheritedFromParent",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "IsViewable",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "DisplayName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ProductCode",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "Email",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "PreHeader",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Folder",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CategoryID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Subject",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsActive",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "IsHTMLPaste",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "ClonedFromID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Status",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "EmailType",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CharacterSet",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "HasDynamicSubjectLine",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "ContentCheckStatus",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.PartnerClientKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ContentAreas",
    			"FieldType": "Text",
    			"SoapType": "ContentArea[]",
    			"IsComplexType": true
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "TextBody",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "HTMLBody",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "EmailSendDefinition",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CategoryID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "SendClassification.CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 128,
    			"IsComplexType": false
    		}, {
    			"Name": "SenderProfile.CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 128,
    			"IsComplexType": false
    		}, {
    			"Name": "SenderProfile.FromName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 255,
    			"IsComplexType": false
    		}, {
    			"Name": "SenderProfile.FromAddress",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 254,
    			"IsComplexType": false
    		}, {
    			"Name": "DeliveryProfile.CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 128,
    			"IsComplexType": false
    		}, {
    			"Name": "DeliveryProfile.SourceAddressType",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 128,
    			"IsComplexType": false
    		}, {
    			"Name": "DeliveryProfile.PrivateIP",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 128,
    			"IsComplexType": false
    		}, {
    			"Name": "DeliveryProfile.DomainType",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 128,
    			"IsComplexType": false
    		}, {
    			"Name": "DeliveryProfile.PrivateDomain",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 128,
    			"IsComplexType": false
    		}, {
    			"Name": "SuppressTracking",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "IsSendLogging",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Email.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "CCEmail",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 254,
    			"IsComplexType": false
    		}, {
    			"Name": "BccEmail",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 254,
    			"IsComplexType": false
    		}, {
    			"Name": "AutoBccEmail",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 254,
    			"IsComplexType": false
    		}, {
    			"Name": "TestEmailAddr",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 254,
    			"IsComplexType": false
    		}, {
    			"Name": "EmailSubject",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 1024,
    			"IsComplexType": false
    		}, {
    			"Name": "DynamicEmailSubject",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsMultipart",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "IsWrapped",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "SendLimit",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "DeduplicateByEmail",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "ExclusionFilter",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Additional",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "SendDefinitionList",
    			"FieldType": "Text",
    			"SoapType": "SendDefinitionList[]",
    			"IsComplexType": true
    		}, {
    			"Name": "IsPlatformObject",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "FileTrigger",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "ExternalReference",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Type",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Status",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "StatusMessage",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "RequestParameterDetail",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ResponseControlManifest",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "FileName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "LastPullDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ScheduledDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "IsActive",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.CreatedBy",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ModifiedBy",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "FileTriggerTypeLastPull",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "ExternalReference",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Type",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "LastPullDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "FilterDefinition",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 128,
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "DataSource.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "DataSource.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "DataFilter",
    			"FieldType": "Text",
    			"SoapType": "FilterPart",
    			"IsComplexType": true
    		}]
    	}, {
    		"objectname": "GlobalUnsubscribeCategory",
    		"isretrievable": true,
    		"candescribe": false,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IgnorableByPartners",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Ignore",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "Group",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Category",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.PartnerClientKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "ImportDefinition",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "FileSpec",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "AllowErrors",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "FieldMappingType",
    			"FieldType": "Text",
    			"SoapType": "ImportDefinitionFieldMappingType",
    			"IsComplexType": true
    		}, {
    			"Name": "FileType",
    			"FieldType": "Text",
    			"SoapType": "FileType",
    			"IsComplexType": true
    		}, {
    			"Name": "UpdateType",
    			"FieldType": "Text",
    			"SoapType": "ImportDefinitionUpdateType",
    			"IsComplexType": true
    		}, {
    			"Name": "MaxFileAge",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "MaxFileAgeScheduleOffset",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "MaxImportFrequency",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "DestinationObject.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "DestinationObject.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "RetrieveFileTransferLocation.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "Delimiter",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "HeaderLines",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "EndOfLineRepresentation",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "NullRepresentation",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "StandardQuotedStrings",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "DateFormattingLocale.LocaleCode",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "ImportResultsSummary",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "ImportDefinitionCustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "TaskResultID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ImportStatus",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "StartDate",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "EndDate",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "DestinationID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "NumberSuccessful",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "NumberDuplicated",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "NumberErrors",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "TotalRows",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ImportType",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "LinkSend",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "SendID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.PartnerClientKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Link.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Link.PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Link.TotalClicks",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Link.UniqueClicks",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Link.URL",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Link.Alias",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "List",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.PartnerClientKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ListName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Category",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Type",
    			"FieldType": "Text",
    			"SoapType": "ListTypeEnum",
    			"IsComplexType": true
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ListClassification",
    			"FieldType": "Text",
    			"SoapType": "ListClassificationEnum",
    			"IsComplexType": true
    		}, {
    			"Name": "AutomatedEmail.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "ListAttribute",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "FieldType",
    			"FieldType": "Text",
    			"SoapType": "ListAttributeFieldType",
    			"IsComplexType": true
    		}, {
    			"Name": "FieldLength",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Scale",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "MinValue",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "MaxValue",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "DefaultValue",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsNullable",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Inheritable",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Overridable",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "MustOverride",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "OverrideType",
    			"FieldType": "Text",
    			"SoapType": "OverrideType",
    			"IsComplexType": true
    		}, {
    			"Name": "List.ObjectID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "List.CustomerKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsHidden",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "IsReadOnly",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Ordinal",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "RestrictedValues",
    			"FieldType": "Text",
    			"SoapType": "ListAttributeRestrictedValue[]",
    			"IsComplexType": true
    		}]
    	}, {
    		"objectname": "ListSend",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "SendID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "List.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "List.ListName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 255,
    			"IsComplexType": false
    		}, {
    			"Name": "Duplicates",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "InvalidAddresses",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ExistingUndeliverables",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ExistingUnsubscribes",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "HardBounces",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "SoftBounces",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "OtherBounces",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ForwardedEmails",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "UniqueClicks",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "UniqueOpens",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "NumberSent",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "NumberDelivered",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Unsubscribes",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "MissingAddresses",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "PreviewURL",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 1024,
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "ListSubscriber",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "SubscriberKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.PartnerClientKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ListID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Status",
    			"FieldType": "Text",
    			"SoapType": "SubscriberStatus",
    			"IsComplexType": true
    		}, {
    			"Name": "UnsubscribedDate",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}]
    	}, {
    		"objectname": "Portfolio",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "RowObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "CategoryID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "FileName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "DisplayName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "TypeDescription",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsUploaded",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "IsActive",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "FileSizeKB",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ThumbSizeKB",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "FileWidthPX",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "FileHeightPX",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "FileURL",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 1024,
    			"IsComplexType": false
    		}, {
    			"Name": "ThumbURL",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 1024,
    			"IsComplexType": false
    		}, {
    			"Name": "CategoryType",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedBy",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedBy",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedByName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "PrivateIP",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsActive",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "OrdinalID",
    			"FieldType": "Text",
    			"SoapType": "Int16",
    			"IsComplexType": true
    		}, {
    			"Name": "IPAddress",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.PartnerClientKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "Program",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Schedule.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "IsActive",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.CreatedBy",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ModifiedBy",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Status",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.EnterpriseID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "ProgramInstance",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "ProgramID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Schedule.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "IsActive",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "StatusMessage",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "StatusLastUpdate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.CreatedBy",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ModifiedBy",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.EnterpriseID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "Publication",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.PartnerClientKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Category",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "PublicKeyManagement",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Key",
    			"FieldType": "Text",
    			"SoapType": "Byte[]",
    			"IsComplexType": true
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "QueryDefinition",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "QueryText",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "TargetType",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "DataExtensionTarget.Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "DataExtensionTarget.CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "DataExtensionTarget.Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "TargetUpdateType",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "FileType",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "FileSpec",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Status",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "CategoryID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "ReplyMailManagementConfiguration",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "EmailDisplayName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ReplySubdomain",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "EmailReplyAddress",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "DNSRedirectComplete",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "DeleteAutoReplies",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "SupportUnsubscribes",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "SupportUnsubKeyword",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "SupportUnsubscribeKeyword",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "SupportRemoveKeyword",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "SupportOptOutKeyword",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "SupportLeaveKeyword",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "SupportMisspelledKeywords",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "SendAutoReplies",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "AutoReplySubject",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "AutoReplyBody",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ForwardingAddress",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ConversationLifetimeDays",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ConversationLifetimeCycles",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "AnonymousRuleSet.ObjectID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "AnonymousRuleSet.Name",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "AnonymousRuleSet.CustomerKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "AnonymousAckTriggeredSend.ObjectID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "AnonymousAckTriggeredSend.CustomerKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "AnonymousAckTriggeredSend.Name",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "AnonymousAckTriggeredSend.TriggeredSendStatus",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "AnonymousForwardTriggeredSend.ObjectID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "AnonymousForwardTriggeredSend.CustomerKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "AnonymousForwardTriggeredSend.Name",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "AnonymousForwardTriggeredSend.TriggeredSendStatus",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ResponderConversationRuleSet.ObjectID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ResponderConversationRuleSet.Name",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ResponderConversationRuleSet.CustomerKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ResponderConversationAckTriggeredSend.ObjectID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ResponderConversationAckTriggeredSend.CustomerKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ResponderConversationAckTriggeredSend.Name",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ResponderConversationAckTriggeredSend.TriggeredSendStatus",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ResponderConversationForwardTriggeredSend.ObjectID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ResponderConversationForwardTriggeredSend.CustomerKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ResponderConversationForwardTriggeredSend.Name",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ResponderConversationForwardTriggeredSend.TriggeredSendStatus",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "InitiatorConversationRuleSet.ObjectID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "InitiatorConversationRuleSet.Name",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "InitiatorConversationRuleSet.CustomerKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "InitiatorConversationAckTriggeredSend.ObjectID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "InitiatorConversationAckTriggeredSend.CustomerKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "InitiatorConversationAckTriggeredSend.Name",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "InitiatorConversationAckTriggeredSend.TriggeredSendStatus",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "InitiatorConversationForwardTriggeredSend.ObjectID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "InitiatorConversationForwardTriggeredSend.CustomerKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "InitiatorConversationForwardTriggeredSend.Name",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "InitiatorConversationForwardTriggeredSend.TriggeredSendStatus",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ConversationExpirationTriggeredSend.ObjectID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ConversationExpirationTriggeredSend.CustomerKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ConversationExpirationTriggeredSend.Name",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ConversationExpirationTriggeredSend.TriggeredSendStatus",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "MultiUseViolationTriggeredSend.ObjectID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "MultiUseViolationTriggeredSend.CustomerKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "MultiUseViolationTriggeredSend.Name",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "MultiUseViolationTriggeredSend.TriggeredSendStatus",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "InboundAddressIsOneUse",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "Send",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.PartnerClientKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Email.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Email.PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "SendDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "FromAddress",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 254,
    			"IsComplexType": false
    		}, {
    			"Name": "FromName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 254,
    			"IsComplexType": false
    		}, {
    			"Name": "Duplicates",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "InvalidAddresses",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ExistingUndeliverables",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ExistingUnsubscribes",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "HardBounces",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "SoftBounces",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "OtherBounces",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ForwardedEmails",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "UniqueClicks",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "UniqueOpens",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "NumberSent",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "NumberDelivered",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "NumberTargeted",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "NumberErrored",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "NumberExcluded",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Unsubscribes",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "MissingAddresses",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Subject",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "PreviewURL",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 1024,
    			"IsComplexType": false
    		}, {
    			"Name": "SentDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "EmailName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 254,
    			"IsComplexType": false
    		}, {
    			"Name": "Status",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"MaxLength": 50,
    			"IsComplexType": false
    		}, {
    			"Name": "IsMultipart",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "SendLimit",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "IsAlwaysOn",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Additional",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "BCCEmail",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "EmailSendDefinition.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "EmailSendDefinition.CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "SendAdditionalAttribute",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "EmailID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "SendID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
                "IsRequired": true,
    			"IsPrimaryKey": true
    		}, {
    			"Name": "AttributeName",
    			"FieldType": "Text",
    			"SoapType": "String",
                "MaxLength": 128,
    			"IsComplexType": false,
                "IsRequired": true,
    			"IsPrimaryKey": true
    		}, {
    			"Name": "AttributeValue",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "SendClassification",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "SendClassificationType",
    			"FieldType": "Text",
    			"SoapType": "SendClassificationTypeEnum",
    			"IsComplexType": true
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "SenderProfile.CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "SenderProfile.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "DeliveryProfile.CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "DeliveryProfile.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "ArchiveEmail",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.PartnerClientKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "SenderProfile",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "FromName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "FromAddress",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "UseDefaultRMMRules",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "AutoForwardToEmailAddress",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "AutoForwardToName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "DirectForward",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "AutoForwardTriggeredSend.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "AutoReply",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "AutoReplyTriggeredSend.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "SenderHeaderEmailAddress",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "SenderHeaderName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ReplyManagementRuleSet.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "RMMRuleCollection.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "SuppressionListContext",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Definition.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "Definition.Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Definition.CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Definition.Category",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Definition.Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Context",
    			"FieldType": "Text",
    			"SoapType": "SuppressionListContextEnum",
    			"IsComplexType": true
    		}, {
    			"Name": "SendClassification.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "Send.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "SenderProfile.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "SendClassificationType",
    			"FieldType": "Text",
    			"SoapType": "SendClassificationTypeEnum",
    			"IsComplexType": true
    		}, {
    			"Name": "Client.CreatedBy",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ModifiedBy",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.EnterpriseID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "AppliesToAllSends",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "SuppressionListDefinition",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.CreatedBy",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ModifiedBy",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Category",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.EnterpriseID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "SubscriberCount",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "Task",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Program.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Sequence",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.UserID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.CreatedBy",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ModifiedBy",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.EnterpriseID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "TaskInstance",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "TaskDefinition.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "Program.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "ProgramInstance.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Sequence",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.UserID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.CreatedBy",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ModifiedBy",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.EnterpriseID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "Template",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "TemplateName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "LayoutHTML",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "BackgroundColor",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "BorderColor",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "BorderWidth",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Cellpadding",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Cellspacing",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Width",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Align",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "ActiveFlag",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "CategoryID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "CategoryType",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "OwnerID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "HeaderContent.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "HeaderContent.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "Layout.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Layout.LayoutName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "TemplateSubject",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsTemplateSubjectLocked",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "TimeZone",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "TriggeredSendDefinition",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CreatedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "ModifiedDate",
    			"FieldType": "Date",
    			"SoapType": "DateTime",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Email.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "List.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Name",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Description",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "TriggeredSendType",
    			"FieldType": "Text",
    			"SoapType": "TriggeredSendTypeEnum",
    			"IsComplexType": true
    		}, {
    			"Name": "TriggeredSendStatus",
    			"FieldType": "Text",
    			"SoapType": "TriggeredSendStatusEnum",
    			"IsComplexType": true
    		}, {
    			"Name": "HeaderContentArea.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "FooterContentArea.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "SendClassification.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "SendClassification.CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "SenderProfile.CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "SenderProfile.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "DeliveryProfile.CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "DeliveryProfile.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "PrivateDomain.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "PrivateIP.ID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "AutoAddSubscribers",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "AutoUpdateSubscribers",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "BatchInterval",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "FromName",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "FromAddress",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "BccEmail",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "EmailSubject",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "DynamicEmailSubject",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsMultipart",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "IsWrapped",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "TestEmailAddr",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "AllowedSlots",
    			"FieldType": "Text",
    			"SoapType": "Int16",
    			"IsComplexType": true
    		}, {
    			"Name": "NewSlotTrigger",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "SendLimit",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "SuppressTracking",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "Keyword",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "List.PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Email.PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "SendClassification.PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "PrivateDomain.PartnerKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "PrivateIP.PartnerKey",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.PartnerClientKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "IsPlatformObject",
    			"FieldType": "Boolean",
    			"SoapType": "Boolean",
    			"IsComplexType": false
    		}, {
    			"Name": "CategoryID",
    			"FieldType": "Number",
    			"SoapType": "Int32",
    			"IsComplexType": false
    		}]
    	}, {
    		"objectname": "TriggeredSendSummary",
    		"isretrievable": true,
    		"candescribe": true,
    		"properties": [{
    			"Name": "ObjectID",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false,
    			"MaxLength": 40,
    			"IsPrimaryKey": true,
    			"IsRequired": true
    		}, {
    			"Name": "Client.ID",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Client.PartnerClientKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "CustomerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "PartnerKey",
    			"FieldType": "Text",
    			"SoapType": "String",
    			"IsComplexType": false
    		}, {
    			"Name": "Sent",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "NotSentDueToOptOut",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "NotSentDueToUndeliverable",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Bounces",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Opens",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "UniqueOpens",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Clicks",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "UniqueClicks",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "OptOuts",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "SurveyResponses",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "FTAFRequests",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "FTAFEmailsSent",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "FTAFOptIns",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "Conversions",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "UniqueConversions",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "InProcess",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "NotSentDueToError",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}, {
    			"Name": "RowObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "TriggeredSendDefinition.ObjectID",
    			"FieldType": "Text",
    			"SoapType": "Guid",
    			"MaxLength": 40,
    			"IsComplexType": false
    		}, {
    			"Name": "Queued",
    			"FieldType": "Number",
    			"SoapType": "Int64",
    			"IsComplexType": false
    		}]
    	}]
    };
}
